src/Components/FilterControls.js

```diff
@@ -1,86 +1,79 @@
     this.state = {
       location: '',
-      time: '',
+      dayOrNight: '',
       date: ''
     }
   }
 
-  getClickedLocation = (event) => {
-    let selectedLocation = event.target.value;
+  getClickedValue = (event) => {
     this.setState({
-      location: selectedLocation
-    }, this.filterByLocation)
+      [event.target.id]: event.target.value
+    }, this.filterParades)
   }
 
-  
-  filterByLocation = () => {
-    return this.props.parades.filter(parade => {
-      return parade.location === this.state.location
-    })
-  }
-  
-  getClickedTime = (event) => {
-    let selectedTime = event.target.value;
-    this.setState({
-      time: selectedTime
-    }, this.filterByTime)
-  }
-  
-  filterByTime = () => {
-    return this.props.parades.filter(parade => {
-      return parade.dayOrNight === this.state.time
+  filterParades = () => { 
+    let keyArr = ['location', 'dayOrNight', 'date']
+    let filteredParades = this.props.parades.filter(parade => {
+      return keyArr.every(key => {
+        return this.state[key] === parade[key] || this.state[key] === ''
+      })
     })
+    return filteredParades
   }
 
-  getClickedDate = (event) => {
-    let selectedDate = event.target.value;
+  resetFilter = () => {
     this.setState({
-      date: selectedDate
-    }, this.filterByDate)
-  }
-
-  filterByDate = () => {
-    return this.props.parades.filter(parade => {
-      return parade.date === this.state.date
+      location: '',
+      dayOrNight: '',
+      date: ''
     })
   }
 
-
-
-
-
   render() {
     return (
       <div>
+        <div className ="filters-section">
+          <div className="buttons">
+              <button className="filter-button"onClick={this.showAllParades}>Show All</button>
+              <button className="filter-button"onClick={this.resetFilter}>Reset Filters</button>
+            </div>
+              <div className="filters">
+                <select className="filter" id="location" onChange={this.getClickedValue}>
+                  <option value="">--Select Location--</option>
+                  <option value="Metairie">Metairie</option>
+                  <option value="Uptown New Orleans">Uptown New Orleans</option>
+                </select>
+                <select className="filter" id="dayOrNight" onChange={this.getClickedValue}>
+                  <option value="">--Select Time--</option>
+                  <option value="am">AM</option>
+                  <option value="pm">PM</option>
+                </select>
+              <select className="filter" id="date" onChange={this.getClickedValue}>
+                <option value="">--Select Date--</option>
+                {this.props.parades.reduce((acc, currentParade) => {
+                  if(!acc.includes(currentParade.date)){acc.push(currentParade.date)}
+                    return acc
+                  },[]).map((name)=> {
+                    return (<option value={[name]}>{[name]}</option>)
+                  })
```

Comment:
Try not to do too much JS logic inside of your render (or at least, not inside of your JSX) -- I would move this out into a variable at the top of the render method, or create a new custom method that returns the JSX you need so you could then replace all this logic with something like `this.generateOptions()`